src/Chapter1.hs

```diff
@@ -542,7 +543,15 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x < y && y < z = y
+    | z < y && y < x = y
+    | y < x && x < z = x
+    | z < x && x < y = x
+    | x < z && z < y = z
+    | y < z && z < x = z
+    | otherwise = 1
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive` if you don't add dummy `otherwise = 1` (even knowing that it never comes to this), as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you added another guard – `| otherwise = 1`, to tell the compiler, that your pattern matching is exhaustive 🙂 However, as you know that the previous cases cover everything, you can move all these 3 cases under a single otherwise and return z in here.
The implementation will be a bit shorter 🙂
```suggestion
    | otherwise = z
```

Reply:
This is clever, thanks!